src/main/java/io/jenkins/plugins/folderauth/FolderBasedAuthorizationStrategy.java

```diff
@@ -113,18 +127,26 @@ public GlobalAclImpl getRootACL() {
     }
 
     /**
-     * Used to initialize transient fields when loaded from disk
+     * Initializes the {@link FolderBasedAuthorizationStrategy} when read from the disk.
      *
-     * @return {@code this}
+     * @return a fully resolved {@link FolderBasedAuthorizationStrategy}
      */
     @Nonnull
     @SuppressWarnings("unused")
-    protected Object readResolve() {
-        jobAcls = new ConcurrentHashMap<>();
-        initCache();
-        generateNewGlobalAcl();
-        updateJobAcls(true);
-        return this;
+    private FolderBasedAuthorizationStrategy readResolve() {
+        return new FolderBasedAuthorizationStrategy(globalRoles, folderRoles);
```

Comment:
Note that this idiom is only safe for immutable classes, as you could accidentally lose customization to mutable persistent fields. Generally prefer the idiom of factoring initialization code into a `private` method called from the constructor and `readResolve`, then having `readResolve` return `this`.